### PR TITLE
Avoid requeue on ClusterQueue status update

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Kueue updates ClusterQueues to include pending or admitted workloads, among other things. These changes shouldn't trigger requeueing.

This not only saves CPU cycles, but API calls to the apiserver for Events.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Skip requeueing of Workloads when there is a status update for a ClusterQueue, saving on API calls for Workloads that were already attempted for admission.
```

### CODE REVIEW

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -615,13 +615,14 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
 			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
 		})
-		ginkgo.It("Should re-enqueue by the update event of ClusterQueue", func() {
+		ginkgo.FIt("Should re-enqueue by the update event of ClusterQueue", func() {
```
Comment (tenzen-y) [inline test/integration/scheduler/scheduler_test.go:618]:
```suggestion
		ginkgo.It("Should re-enqueue by the update event of ClusterQueue", func() {
```

Debug code still remain :)

**Review by tenzen-y — COMMENTED** [review]
I have only reviewed some of the codes yet since CI failed.

File: pkg/queue/cluster_queue_impl.go [hunk 2]
```diff
@@ -119,6 +119,7 @@ func (c *clusterQueueBase) PushOrUpdate(wInfo *workload.Info) {
 		// to potentially become admissible, unless the Eviction status changed
 		// which can affect the workloads order in the queue.
 		if equality.Semantic.DeepEqual(oldInfo.Obj.Spec, wInfo.Obj.Spec) &&
+			equality.Semantic.DeepEqual(oldInfo.Obj.Status.ReclaimablePods, wInfo.Obj.Status.ReclaimablePods) &&
```
Comment (tenzen-y) [inline pkg/queue/cluster_queue_impl.go:122]: Should we extend this UT: https://github.com/kubernetes-sigs/kueue/blob/1e66b4360a9870136c816372a5818f35b50bf31b/pkg/queue/cluster_queue_impl_test.go#L228

**Review by tenzen-y — COMMENTED** [review]
Generally lgtm

File: pkg/queue/cluster_queue_impl.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/queue/cluster_queue_impl.go:122]: added

File: pkg/controller/core/workload_controller.go [hunk 3]
```diff
@@ -541,9 +542,38 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
...
+		if wlCopy.Status.RequeueState != nil && wlCopy.Status.RequeueState.RequeueAt != nil {
+			backoff = time.Until(wl.Status.RequeueState.RequeueAt.Time)
 		}
+		if backoff <= 0 {
+			if !r.queues.AddOrUpdateWorkload(wlCopy) {
+				log.V(2).Info("Queue for workload didn't exist; ignored for now")
+			}
+		} else {
+			log.V(3).Info("Workload to be requeued after backoff", "backoff", backoff, "requeueAt", wl.Status.RequeueState.RequeueAt.Time)
+			time.AfterFunc(backoff, func() {
```
Comment (tenzen-y) [inline pkg/controller/core/workload_controller.go:555]:
```suggestion
			time.AfterFunc(_, func() {
```

It seems that the `backoff` has never been used.

**Review by tenzen-y — COMMENTED** [review]
Ideally, we should extend UTs, but integraiton tests should be enough now.

File: pkg/controller/core/workload_controller.go (hunk 3 above)
Comment (tenzen-y) [reply pkg/controller/core/workload_controller.go:555]: Oops, this is necessary for `AfterFunc`.
NVM

**Comment by tenzen-y** [comment]
Thank you!
/lgtm
/approve

**Pull request merged**